{{cookiecutter.project_dirname}}/scripts/report.sh

```diff
@@ -4,4 +4,4 @@ set -euo pipefail
 
 python3 -m coverage combine
 python3 -m coverage html
-python3 -m coverage report
+python3 -m coverage report --fail-under=100
```

Comment:
This seems like an unnecessary move to me, because the configurations on the pyproject.toml are unique to the coverage report command and as such I would expect it to work no matter where the fail-under=100 argument is placed.
If the report is not printed it is very likely that the set -euo pipefail that we have in the bash script report.sh interrupts the printing of the repo at an unnecessary point.
Personally I would leave the fail under configuration in the pyproject.toml and investigate a possible modification of the report.sh pipefail.

Reply:
change `report.sh` with `set -uo pipefail` (without `-e`) could be the way

Reply:
That was the first solution I thought about, and it worked. Unfortunately, the `test.sh` script also launches the `check.sh` script, which is supposed to fail and block the execution. Therefore, the halting policy is not homogeneous in that script, and that is why I proposed to only set the failing criterium on the last coverage command. An alternative could be to move out the `check.sh` invocation. Whatever works for everybody, but let's quickly solve this issue, so that the report is printed also when (or especially when) a low coverage causes a failure.

Reply:
I don't understand the connection with the `check.sh` script, if the check fails it won't print the coverage report anyway

however, with this change, homogeneity is not respected in the configurations of the python tools, we choose which to prefer, it seems to me that it makes more sense to configure the `pipefail` of the sh scripts with what they are expected to do

Reply:
If the check fails, and the `-e` option is missing, the execution keeps going, which is useless and time consuming. In case of a check failure, execution should stop.